### PR TITLE
Looker

### PR DESCRIPTION
**Pull request by BenJoParadise**
## Status
Awaiting customer feedback

## Related Issues
fixes: https://github.com/demisto/etc/issues/16097

## Description
Integration with Looker

## Does it break backward compatibility?
   - No

## Must have
- [X] Tests
- [x] Documentation - https://github.com/demisto/etc/issues/17120
- [x] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [ ] All tests

## Additional changes
Fixed `camelize()` function in `CommonServerPython`

### CODE REVIEW

**Comment by BenJoParadise** [comment]
@kirbles19 - Ready for documentation review:
yaml - https://github.com/demisto/content/blob/f54d91545e2d414cfeca3beee00564467c6998ad/Integrations/Looker/Looker.yml
description - https://github.com/demisto/content/blob/f54d91545e2d414cfeca3beee00564467c6998ad/Integrations/Looker/Looker_description.md

**Comment by kirbles19** [comment]
@BenJoParadise Is the .yml and .md ready for me to review?

**Comment by BenJoParadise** [comment]
@kirbles19 Yes, i will open an issue for zendesk docs.

File: Integrations/Looker/Looker.py [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+    HEADERS['Authorization'] = 'token {}'.format(ic[CLIENT_ID]['token'])
+
+
+def get_limit():
+    try:
+        limit = int(demisto.args().get('limit', DEFAULT_RESULTS_LIMIT))
+        return None if limit == 0 else limit
+
+    except ValueError:
+        return_error("limit must be a number")
```
Comment (BenJoParadise) [inline Integrations/Looker/Looker.py]: raise

File: Integrations/Looker/Looker.py [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+    }
+    response_json = http_request('POST', '/login', data=data)
+
+    return {
+        'token': response_json['access_token'],
+        'expires': datetime.utcnow().timestamp() + response_json['expires_in']
+    }
+
+
+def get_session_token():
```
Comment (BenJoParadise) [inline Integrations/Looker/Looker.py]: pass global params

File: Integrations/Looker/Looker.py [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+    result_format = demisto.args()['result_format']
+    limit = get_limit()
+    fields = argToList(demisto.args().get('result_format'))
+
+    contents = run_look_request(look_id, result_format, limit, fields)
+
+    demisto.results(get_entries_for_search_results(contents, look_id, result_format, look_name))
+
+
+def run_look_request(look_id, result_format, limit, fields):
```
Comment (BenJoParadise) [inline Integrations/Looker/Looker.py]: remove request

File: Integrations/Looker/Looker.py [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+    if look_name and not look_id:
+        look_id = get_look_id_from_name(look_name)
+
+    result_format = demisto.args()['result_format']
+    limit = get_limit()
+    fields = argToList(demisto.args().get('result_format'))
+
+    contents = run_look_request(look_id, result_format, limit, fields)
+
+    demisto.results(get_entries_for_search_results(contents, look_id, result_format, look_name))
```
Comment (BenJoParadise) [inline Integrations/Looker/Looker.py]: return outputs

File: Integrations/Looker/Looker.py [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+            )
+
+    elif result_format == 'csv':
+        entries.append(fileResult('look_result.csv' if look_id else 'inline_query_result.csv', contents,
+                                  entryTypes['entryInfoFile']))
+
+    return entries
+
+
+def get_query_args():
```
Comment (BenJoParadise) [inline Integrations/Looker/Looker.py]: pass demisto.args() as method argument

Comment (BenJoParadise) [reply Integrations/Looker/Looker.py]: Add unitest

File: Integrations/Looker/Looker.py [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+        create_look_command()
+
+# Log exceptions
+except Exception as e:
+    LOG(e)
+    LOG(traceback.format_exc())
+    LOG.print_log()
+    if demisto.command() != 'test-module':
+        return_error(str(e))
+    demisto.results(e)
```
Comment (BenJoParadise) [inline Integrations/Looker/Looker.py]: flip the if, add else

File: Integrations/Looker/Looker.py [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+
+    return res_obj
+
+
+def get_new_token():
+    data = {
+        'client_id': CLIENT_ID,
+        'client_secret': CLIENT_SECRET
+    }
+    response_json = http_request('POST', '/login', data=data)
```
Comment (BenJoParadise) [inline Integrations/Looker/Looker.py]: catch 404, add invalid credentials message

File: Integrations/Looker/Looker.yml [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+commonfields:
+  id: Looker
+  version: -1
+configuration:
+- display: API URL and port (e.g. https://example.looker.com:19999)
+  name: url
+  required: true
+  type: 0
+- defaultvalue: 'true'
+  display: Trust any certificate (unsecure)
```
Comment (Itay4) [inline Integrations/Looker/Looker.yml]: change to `not secure`

File: Integrations/Looker/Looker.yml [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+  name: unsecure
+  required: false
+  type: 8
+- display: Use system proxy
```
Comment (Itay4) [inline Integrations/Looker/Looker.yml]: change to `Use system proxy settings`

**Comment by BenJoParadise** [comment]
Ready to merge

**Comment by BenJoParadise** [comment]
@anara123 FYI, I've added a missing method to CommonServerPython since your last review.
It is now tested and ready to merge.

**Comment by anara123** [comment]
Very weird circle build. 20 failed tests when in master only 3-4 unstable tests. And most of the tests failed on not provided credentials

**Comment by anara123** [comment]
need to check the root cause for that before we merge

**Comment by BenJoParadise** [comment]
continued in: https://github.com/demisto/content/pull/3649

**Pull request closed without merging**